src/libraries/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs

```diff
@@ -258,23 +268,46 @@ public static DirectoryInfo CreateDirectory(this DirectorySecurity directorySecu
 
         // In the context of a FileStream, the only ACCESS_MASK ACE rights we care about are reading/writing data and the generic read/write rights.
         // See: https://docs.microsoft.com/en-us/windows/win32/secauthz/access-mask
-        private static FileAccess GetFileStreamFileAccess(FileSystemRights rights)
+        private static FileAccess GetFileAccessFromRights(FileSystemRights rights)
```

Comment:
This is the new code where `FileSystemRights` are mapped to a `FileAccess` value.

In .NET Framework, this is where the `fAccess` value is passed to the P/Invoke:
https://referencesource.microsoft.com/#mscorlib/system/io/filestream.cs,845

And here is where the value was simply casted to `int`.
https://referencesource.microsoft.com/#mscorlib/system/io/filestream.cs,705

In our case, we need to pass it as a `FileAccess`.